Sources/MessagingPush/MessagingPush+AppDelegate.swift

```diff
@@ -65,7 +65,27 @@ extension MessagingPushImplementation {
         switch response.actionIdentifier {
         case UNNotificationDefaultActionIdentifier: // push notification was touched.
             if let deepLinkurl = pushContent.deepLink {
-                UIApplication.shared.open(url: deepLinkurl)
+                // First, try to open the link inside of the host app. This is to keep compatability with Universal Links.
+                // Learn more of edge case: https://github.com/customerio/customerio-ios/issues/262
+                // Fallback to opening the URL system-wide if fail to open link in host app.
+                // Customers with Universal Links in their app will need to add this function to their `AppDelegate` which will get called with deep link:
+                // func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool
+
```

Comment:
I saw there is already a method available `UIApplication.shared.canOpenURL`. Can we utilize it maybe? 

```suggestion
if UIApplication.shared.canOpenURL(deepLinkurl) {
            UIApplication.shared.open(deepLinkurl, options: [:], completionHandler: nil)
            return true
        } else {
            // Fall back to opening the URL system-wide
            UIApplication.shared.open(deepLinkurl, options: [:], completionHandler: nil)
            return false
        }
```

Reply:
I previously tried `canOpenURL` but it doesn't solve the problem we are having. At least, `canOpenURL` doesn't work for universal links. It might work for app schemes, but we are not experiencing any issues with app schemes at the moment.